src/client/app/containers/LineChartContainer.ts

```diff
@@ -31,19 +24,49 @@ function mapStateToProps(state: State) {
 					throw new Error('Unacceptable condition: readingsData.readings is undefined.');
 				}
 
-				const dataPoints: ChartPoint[] = _.values(readingsData.readings).map(compressedReadingToChartPoint);
+				// Create two arrays for the x and y values. Fill the array with the data from the compressed readings
+				const xData: string[] = [];
+				const yData: number[] = [];
+				const hoverText: string[] = [];
+				const readings = _.values(readingsData.readings);
+				readings.forEach(reading => {
+					const timeReading = moment(reading.startTimestamp);
+					xData.push(timeReading.format('YYYY-MM-DD HH:mm:ss'));
+					yData.push(reading.reading);
+					hoverText.push(`<b> ${timeReading.format('dddd, MMM DD, YYYY hh:mm a')} </b> <br> ${label}: ${reading.reading} kW`);
+				});
+
+				// Save the timestamp range of the plot
+				let minTimestamp: string = "";
+				let maxTimestamp: string = "";
+				if (readings.length > 0){
+					minTimestamp = readings[0]["startTimestamp"].toString();
+					maxTimestamp = readings[readings.length-1]["startTimestamp"].toString();
+				}
+				let root: any = document.getElementById('root');
+				root.setAttribute("min-timestamp", minTimestamp);
+				root.setAttribute("max-timestamp", maxTimestamp);
 
+				// This variable contains all the elements (x and y values, line type, etc.) assigned to the data parameter of the Plotly object
 				datasets.push({
-					label,
-					data: dataPoints,
-					fill: false,
-					borderColor: getGraphColor(label)
+					name: label,
+					x: xData,
+					y: yData,
+					text: hoverText,
+					hoverinfo: 'text',
+					type: 'scatter',
+					mode: 'lines',
+					line: {
+						shape: 'spline',
+						width: 3
+					},
+					marker: {color: getGraphColor(label)}
 				});
 			}
 		}
 	}
 
```

Comment:
```suggestion

// TODO The meters and groups code is very similar and maybe it should be refactored out to create a function to do 
// both. This would mean future changes would automatically happen to both.
```